app/models/place.rb

```diff
@@ -448,39 +439,40 @@ def validate_with_geom( geom, other_attrs = {} )
       add_custom_error( :base, "Failed to import a boundary. Check for slivers, overlapping polygons, and other geometry issues." )
       return false
     end
-    observation_count = Observation.where("ST_Intersects(private_geom, ST_GeomFromEWKT(?))", geom.as_text).count
-    if other_attrs[:user] && !other_attrs[:user].is_admin?
-      if geom.respond_to?(:area) && (
-         ( CONFIG.content_freeze_enabled && geom.area > MAX_PLACE_AREA_FOR_NON_STAFF_DURING_FREEZE ) ||
-         geom.area > MAX_PLACE_AREA_FOR_NON_STAFF )
-        add_custom_error( :place_geometry, :is_too_large_to_import )
-        return false
-      elsif ( ( CONFIG.content_freeze_enabled && observation_count >= MAX_PLACE_OBSERVATION_COUNT_DURING_FREEZE ) ||
-        observation_count >= MAX_PLACE_OBSERVATION_COUNT )
+
+    if max_observation_count
+      observation_count = Observation.where("ST_Intersects(private_geom, ST_GeomFromEWKT(?))", geom.as_text).count
+      if observation_count > max_observation_count
         add_custom_error(:place_geometry, :contains_too_many_observations)
         return false
       end
     end
+
+    if max_area_km2
+      area_km2 = PlaceGeometry.area_km2( geom )
+      if area_km2 > max_area_km2
+        add_custom_error( :place_geometry, :is_too_large_to_import )
+        return false
+      end
+    end
+
     true
   end
   
   # Update the associated place_geometry or create a new one
-  def save_geom( geom, other_attrs = {} )
+  def save_geom( geom, max_area_km2: nil, max_observation_count: nil, **other_attrs )
```

Comment:
I don't have a problem with this approach, but it's still causing tests to fail. They pass for me if I change this to

```ruby
def save_geom( geom, other_attrs = {}, max_area_km2: nil, max_observation_count: nil )
```

Reply:
Thanks. Addressed in 40a7cdc47. I need to get less lazy about picking tests to run locally.